### PR TITLE
Update config.ex to allow sdk users to retrieve the project id.

### PR DESCRIPTION
**Pull request by mmaughon**
This change prevents the GenServer from crashing for SDK users when the project_id is not available.

After some investigation, if the project id is not set in the dynamic_config or through one of the following environment variables(GOOGLE_CLOUD_PROJECT, GCLOUD_PROJECT, DEVSHELL_PROJECT_ID) then a request is sent to Google to retrieve the project id through metadata. If the request fails, the GenServer crashes. However, if the request completes, even if the response contains bad data, the GenServ will continue to run.

For SDK users, there is a second file that contains the project id and email address. This change will parse the file (configurations/config_default) to have another way of locally obtaining the project id and email address before attempting a request to Google for the project id.

This change performs the following steps:
1) Parse the new configuration file if it's present.
2) Merge the required configuration data with the credential data.
3) Add a check for project id in `determine_project_id` as another method for retrieving the project id.

Also, I was unable to use the TOML module to parse the ini formatted file since it contains a bare email address which is considered invalid TOML because of the ampersand. I wrote a small function that would just parse what was needed to retrieve the project id and email.

Here is the format of my `configurations/config_default`:
[core]
account = user@domain.com
project = my_project
disable_usage_reporting = False

[compute]
zone = us-central1-c
region = us-central1

### CODE REVIEW

**Review by wojtekmach — COMMENTED** [review]
Thank you for the PR and apologies for the delay. Please add a test and we should be good to go.

File: lib/goth/config.ex [hunk 1]
```diff
@@ -195,9 +220,9 @@ defmodule Goth.Config do
   end
 
   defp determine_project_id(config, dynamic_config) do
-    case Keyword.get(dynamic_config, :project_id) || System.get_env("GOOGLE_CLOUD_PROJECT") ||
-           System.get_env("GCLOUD_PROJECT") || System.get_env("DEVSHELL_PROJECT_ID") ||
-           config["project_id"] || config["quota_project_id"] do
+    case Keyword.get(dynamic_config, :project_id) || Map.get(config, :project_id) ||
+           System.get_env("GOOGLE_CLOUD_PROJECT") || System.get_env("GCLOUD_PROJECT") ||
+           System.get_env("DEVSHELL_PROJECT_ID") || config["project_id"] || config["quota_project_id"] do
```
[the remark this answers is not in the document]
Comment (mmaughon) [reply lib/goth/config.ex]: Done. Good catch. I didn't see it the first time.

File: lib/goth/config.ex [hunk 2]
```diff
@@ -167,26 +183,35 @@ defmodule Goth.Config do
...
+      credential_file |> File.read!() |> decode_json()
+    else
+      %{}
+    end
+  end
 
-    path = Path.join([path_root, "gcloud", "application_default_credentials.json"])
+  defp get_configuration_data(configuration_file) do
+    if File.regular?(configuration_file) do
+      configuration_data = configuration_file |> File.read!() |> decode_toml()
```
[the remark this answers is not in the document]
Comment (mmaughon) [reply lib/goth/config.ex]: Done.

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**